src/event_handler.rs

```diff
@@ -504,63 +511,148 @@ pub async fn handle_event_logs(
+                * pending_txn_data.gas_price
+                / 100;
+
+            // Send the replacement txn if the current one is dropped or lost
+            while Instant::now() < pending_txn_data.retry_deadline {
+                let replacement_txn = pending_txn_data.txn_data.clone();
+                let replacement_txn = replacement_txn
+                    .nonce(pending_txn_data.nonce)
+                    .gas_price(pending_txn_data.gas_price)
+                    .gas(pending_txn_data.gas_limit);
+
+                let pending_txn = replacement_txn.send().await;
+                let Ok(pending_txn) = pending_txn else {
+                    let error_string = format!("{:?}", pending_txn.unwrap_err());
+
+                    // Retry after a delay if connection failed
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("connection") || error_string.contains("network"))
+                    {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+
+                    // Retry with a higher gas price if the replacement txn is underpriced
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("replacement transaction underpriced")
+                            || error_string.contains("transaction underpriced"))
+                    {
+                        pending_txn_data.gas_price =
+                            U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                                * pending_txn_data.gas_price
+                                / 100;
+
+                        sleep(Duration::from_millis(10)).await;
+                        continue;
+                    }
+
+                    // Break in case of other errors like "nonce too low" or "execution reverted" or "insufficient funds" etc.
+                    retry_txn = false;
+                    break;
+                };
+
+                // Monitor the newly sent pending txn
+                pending_txn_data.txn_hash = pending_txn.tx_hash();
+                break;
+            }
         }
 
-        // Update gas limit and price for sending the replacement transaction with the original nonce
-        gas_limit += GAS_LIMIT_BUFFER.into();
-        gas_price += (U256::from(RESEND_GAS_PRICE_INCREMENT_PERCENT) * gas_price) / 100;
-
-        // Update replacement transaction metadata and send it to the common chain, stop resending if failed to do so once
-        let replacement_txn = txn.clone();
-        let replacement_txn = replacement_txn
-            .nonce(nonce)
-            .gas(gas_limit)
-            .gas_price(gas_price);
-        let Ok(pending_txn) = replacement_txn.send().await else {
-            resend_retries = 0;
-            break;
-        };
+        while retry_txn == true {
```

Comment:
if while really required here? You have the loop inside it already. It's getting complicated, please simplify it.

Reply:
Yes will simplify the flow.